src/Data/Int.purs

```diff
@@ -77,6 +79,21 @@ foreign import toNumber :: Int -> Number
 fromString :: String -> Maybe Int
 fromString = fromStringAs (Radix 10)
 
+-- | A type for describing whether an integer is even or odd.
+data Parity = Even | Odd
+
+derive instance eqParity :: Eq Parity
+derive instance ordParity :: Ord Parity
```

Comment:
I don't think we should. Just because it has two elements doesn't mean we should think of it as Z/2 in my opinion. I would rather have a separate type.